test/udev-test.py

```diff
@@ -0,0 +1,2408 @@
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda10",
+            not_exp_links   = ["partition"],
+            exp_links       = ["part-10"],
+        ),
+        rules = r"""
+        SUBSYSTEM=="block", SUBSYSTEMS=="scsi", KERNEL=="sda?*", ENV{DEVTYPE}=="partition", SYMLINK+="part-%n"
+        SUBSYSTEM=="block", SUBSYSTEMS=="scsi", KERNEL=="sda?*", ENV{DEVTYPE}=="partition", SYMLINK+="partition"
+        KERNEL!="*7", OPTIONS+="link_priority=-10"
+        """),
+
+    Rules.new(
+        "multiple devices, same link name, positive prio, sleep",
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda1",
+            exp_links       = ["part-1"],
+            not_exp_links   = ["partition"],
+        ),
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda5",
+            exp_links       = ["part-5"],
+            not_exp_links   = ["partition"],
+        ),
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda6",
+            not_exp_links   = ["partition"],
+            exp_links       = ["part-6"],
+        ),
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda7",
+            exp_links       = ["part-7", "partition"],
+        ),
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda8",
+            not_exp_links   = ["partition"],
+            exp_links       = ["part-8"],
+        ),
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda9",
+            not_exp_links   = ["partition"],
+            exp_links       = ["part-9"],
+        ),
+        Device(
+            "/devices/pci0000:00/0000:00:1f.2/host0/target0:0:0/0:0:0:0/block/sda/sda10",
+            not_exp_links   = ["partition"],
+            exp_links       = ["part-10"],
+        ),
+        delay = 10000,
+        rules = r"""
+        SUBSYSTEM=="block", SUBSYSTEMS=="scsi", KERNEL=="sda?*", ENV{DEVTYPE}=="partition", SYMLINK+="part-%n"
+        SUBSYSTEM=="block", SUBSYSTEMS=="scsi", KERNEL=="sda?*", ENV{DEVTYPE}=="partition", SYMLINK+="partition"
+        KERNEL=="*7", OPTIONS+="link_priority=10"
+        """),
+
+    Rules.new(
+        'all_block_devs',
+        *all_block_devs(lambda name: (["blockdev"], None) if name.endswith('/sda6') else (None, None)),
+        repeat = 10,
+        rules  = r"""
+        SUBSYSTEM=="block", SUBSYSTEMS=="scsi", KERNEL=="sd*", SYMLINK+="blockdev"
+        KERNEL=="sda6", OPTIONS+="link_priority=10"
+        """),
+]
+
+def fork_and_run_udev(action: str, rules: Rules) -> None:
+    kinder = []
+    for k, device in enumerate(rules.devices):
+        # TODO: valgrind/gdb/strace
+        cmd = [UDEV_BIN, action, device.devpath]
+        if rules.delay:
+            cmd += [f'{k * rules.delay}']
```

Comment:
Why `k` is multiplied to delay?

Reply:
That's what the original code was doing. AFAUI, it wants to start the jobs staggered.